shared/feature/feeding/src/main/java/com/epmedu/animeal/feeding/domain/usecase/GetFeedingPointByPriorityUseCase.kt

```diff
@@ -0,0 +1,53 @@
+package com.epmedu.animeal.feeding.domain.usecase
+
+import com.epmedu.animeal.feeding.domain.model.FeedingPoint
+import com.epmedu.animeal.feeding.domain.model.enum.AnimalState
+import com.epmedu.animeal.feeding.presentation.model.FeedingPointModel
+import com.epmedu.animeal.feeding.presentation.model.MapLocation
+import com.epmedu.animeal.feeding.presentation.model.MapLocation.Companion.toPoint
+import com.epmedu.animeal.foundation.tabs.model.AnimalType
+import com.epmedu.animeal.geolocation.location.distanceInKmTo
+import com.epmedu.animeal.geolocation.location.isNearTo
+
+class GetFeedingPointByPriorityUseCase(
+    val getAllFeedingPointsUseCase: GetAllFeedingPointsUseCase
+) {
+
+    /*
+     * Filters iterable of FeedingPointModel by priority defined at EPMEDU-594:
+     * - 1st: Status = Red, is favorite and is near to user location
+     * - 2nd: Status = Red, is favorite
+     * - 3rd: Status = Red, is not favorite but is near to user location
+     */
+    suspend operator fun invoke(
+        type: AnimalType,
+        userLocation: MapLocation,
+        onFeedingPointFound: (FeedingPointModel) -> Unit,
+    ) {
+        getAllFeedingPointsUseCase(type = type).collect { domainFeedingPoints ->
+            val readyToFeed = domainFeedingPoints
+                .filter { it.animalStatus == AnimalState.RED }
```

Comment:
We have repository with getFeedingPointsBy(predicate) method, better to modify getAllFeedingPointsUseCase and add support for alternate predicate, or just implement repo method with specified predicates

Reply:
not resolved